pkg/agent/flowexporter/connections/l7_listener.go

```diff
@@ -0,0 +1,212 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connections
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/netip"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/flowexporter"
+
+	k8sutil "antrea.io/antrea/pkg/util/k8s"
+	"antrea.io/antrea/pkg/util/podstore"
+)
+
+type PodL7VisibilityAttrGetter interface {
+	IsL7VisibilityRequested(podNN string, ingress bool) bool
+}
+
+// L7ProtocolFields holds layer 7 protocols supported
+type L7ProtocolFields struct {
+	http []map[int32]*Http
+}
+
+// Http holds the L7 HTTP flow JSON values.
+type Http struct {
+	Hostname      string `json:"hostname"`
+	URL           string `json:"url"`
+	UserAgent     string `json:"http_user_agent"`
+	ContentType   string `json:"http_content_type"`
+	Method        string `json:"http_method"`
+	Protocol      string `json:"protocol"`
+	Status        int32  `json:"status"`
+	ContentLength int32  `json:"length"`
+}
+
+// JsonToEvent holds Suricata event JSON values.
+// See https://docs.suricata.io/en/latest/output/eve/eve-json-format.html?highlight=HTTP%20event#event-types
+type JsonToEvent struct {
+	Timestamp   string     `json:"timestamp"`
+	FlowID      int64      `json:"flow_id"`
+	InInterface string     `json:"in_iface"`
+	EventType   string     `json:"event_type"`
+	VLAN        []int32    `json:"vlan"`
+	SrcIP       netip.Addr `json:"src_ip"`
+	SrcPort     int32      `json:"src_port"`
+	DestIP      netip.Addr `json:"dest_ip"`
+	DestPort    int32      `json:"dest_port"`
+	Proto       string     `json:"proto"`
+	TxID        int32      `json:"tx_id"`
+	HTTP        *Http      `json:"http"`
+}
+
+type L7Listener struct {
+	l7Events                  map[flowexporter.ConnectionKey]L7ProtocolFields
+	l7mut                     sync.Mutex
+	suricataEventSocketPath   string
+	podL7VisibilityAttrGetter PodL7VisibilityAttrGetter
+	podStore                  podstore.Interface
+}
+
+func NewL7Listener(
+	podL7VisibilityAttrGetter PodL7VisibilityAttrGetter,
+	podStore podstore.Interface) *L7Listener {
+	return &L7Listener{
+		l7Events:                  make(map[flowexporter.ConnectionKey]L7ProtocolFields),
+		suricataEventSocketPath:   config.L7SuricataSocketPath,
+		podL7VisibilityAttrGetter: podL7VisibilityAttrGetter,
+		podStore:                  podStore,
+	}
+}
+
+func (l *L7Listener) Run(stopCh <-chan struct{}) {
+	go wait.Until(l.listenAndAcceptConn, 5*time.Second, stopCh)
+	<-stopCh
+}
+
+func (l *L7Listener) listenAndAcceptConn() {
+	// Remove stale connections
+	if err := os.Remove(l.suricataEventSocketPath); err != nil && !os.IsNotExist(err) {
+		klog.V(2).ErrorS(err, "failed to remove stale socket")
+	}
+	listener, err := net.Listen("unix", l.suricataEventSocketPath)
+	if err != nil {
+		klog.V(2).ErrorS(err, "failed to listen on Suricata socket")
```

Comment:
```suggestion
		klog.ErrorS(err, "Failed to listen on Suricata socket")
```
1. Error shouldn't be V(2), otherwise you don't know the problem when there is something wrong.
2. Log should be capitalized first letter.

Please check other places.

Reply:
we may need handling for this otherwise unless we have Suricata running, we will keep getting this log and that will simply flood the logs

Reply:
your enquiry is correct, however when the socket file is missing on the node running Suricata, we receive an error file not found and to not keep printing that error, I added the verbose level 2 for this scenario

Reply:
I can confirm listening a unix socket will create the file. The reason why it met errors here is the directory didn't exist. I think you should ensure the path exist before listening, instead of waiting for suricata to create the directory.